plugins/vpc-shared-eni/config/kubernetes.go

```diff
@@ -50,8 +50,11 @@ var (
 )
 
 // parseKubernetesArgs parses Kubernetes-specific CNI arguments.
-func parseKubernetesArgs(netConfig *NetConfig, args *cniSkel.CmdArgs) error {
-	if args == nil || args.Args == "" {
+func parseKubernetesArgs(netConfig *NetConfig, args *cniSkel.CmdArgs, isAddCmd bool) error {
+	// The only additional information we need to query from API server is the pod IP address,
+	// which is required only for ADD commands. Also the API server may have deleted the pod
+	// object already in the DEL path.
+	if !isAddCmd ||  args == nil || args.Args == "" {
```

Comment:
We could have this block at line 75 (right after populating the kc fields), so that the namespace, podname and podinfracontainerid are available also in the DEL path. I understand they are not currently used, so this is more for completeness rather than functional correctness. Not a big deal.

Reply:
Makes sense, fixed it in the next commit.